### PR TITLE
Precheck plugin deletion

### PR DESCRIPTION
**Pull request by buehner**
Introduce a new web interface that allows to check from which applications a plugin would be removed in case of deletion.

### CODE REVIEW

**Comment by buehner** [comment]
Once https://github.com/terrestris/shogun2/pull/262 is merged, these changes can be merged here to make the build green

**Comment by buehner** [comment]
As this is green now, please review @marcjansen @dnlkoch

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PluginService.java [hunk 1]
```diff
@@ -97,6 +98,30 @@ public String getPluginSource(String simpleClassName) throws Exception {
 	}
 
 	/**
+	 *
+	 * @param pluginId
+	 * @return List of application names that contain the given plugin
+	 */
+	@PreAuthorize("hasRole(@configHolder.getSuperAdminRoleName()) or hasPermission(#layerId, 'de.terrestris.shogun2.model.Plugin', 'DELETE')")
```
Comment (dnlkoch) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PluginService.java]: `#layerId` -> `#pluginId`?

**Comment by dnlkoch** [comment]
One minor remark only. Feel free to merge!

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PluginService.java (hunk 1 above)
Comment (buehner) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PluginService.java]: nice catch! i'll fix it

**Pull request merged**